base/associative.jl

```diff
@@ -35,29 +35,29 @@ function summary(t::Associative)
     return string(typeof(t), " with ", n, (n==1 ? " entry" : " entries"))
 end
 
-struct KeyIterator{T<:Associative}
+struct KeySet{K, T <: Associative{K}} <: AbstractSet{K}
     dict::T
 end
+KeySet(dict::Associative) = KeySet{keytype(dict), typeof(dict)}(dict) 
+
 struct ValueIterator{T<:Associative}
```

Comment:
Shouldn't this to be changed to `ValueSet` at the same time?

Reply:
Unlike keys, values don’t have to be unique, so they aren’t really a set.